core/beacon/src/chain.rs

```diff
@@ -162,6 +162,7 @@ impl<B: Block> BlockChain<B> {
 
         let maybe_parent = self.get_header(&BlockId::Hash(block.header().parent_hash()));
         if let Some(_parent_details) = maybe_parent {
+            // TODO: fork choice rule
```

Comment:
what is the interface for the fork choice rule?
I'm thinking we should make it composable with BlockChain (in #112 I'm moving chain.rs into a separate module) and compose here `BlockChain<BeaconBlock, BeaconForkChoice>`